ui/src/app/machines/components/TakeActionMenu/TakeActionMenu.tsx

```diff
@@ -1,93 +1,141 @@
+    actions: [
+      { label: "Commission...", name: NodeActions.COMMISSION },
+      { label: "Acquire...", name: NodeActions.ACQUIRE },
+      { label: "Deploy...", name: NodeActions.DEPLOY },
+      { label: "Release...", name: NodeActions.RELEASE },
+      { label: "Abort...", name: NodeActions.ABORT },
+    ],
+  },
+  {
+    name: "power",
+    actions: [
+      { label: "Power on...", name: NodeActions.ON },
+      { label: "Power off...", name: NodeActions.OFF },
+    ],
+  },
+  {
+    name: "testing",
+    actions: [
+      { label: "Test...", name: NodeActions.TEST },
+      { label: "Enter rescue mode...", name: NodeActions.RESCUE_MODE },
+      { label: "Exit rescue mode...", name: NodeActions.EXIT_RESCUE_MODE },
+      { label: "Mark fixed...", name: NodeActions.MARK_FIXED },
+      { label: "Mark broken...", name: NodeActions.MARK_BROKEN },
+      {
+        label: "Override failed testing...",
+        name: NodeActions.OVERRIDE_FAILED_TESTING,
+      },
+    ],
+  },
+  {
+    name: "lock",
+    actions: [
+      { label: "Lock...", name: NodeActions.LOCK },
+      { label: "Unlock...", name: NodeActions.UNLOCK },
+    ],
+  },
+  {
+    name: "misc",
+    actions: [
+      { label: "Tag...", name: NodeActions.TAG },
+      { label: "Set zone...", name: NodeActions.SET_ZONE },
+      { label: "Set pool...", name: NodeActions.SET_POOL },
+      { label: "Delete...", name: NodeActions.DELETE },
+    ],
+  },
+];
+
 const getTakeActionLinks = (
-  actionOptions: MachineAction[],
   machines: Machine[],
   setSelectedAction: Props["setSelectedAction"],
   appearance: Props["appearance"]
 ) => {
-  const initGroups = [
-    { type: "lifecycle", items: [] },
-    { type: "power", items: [] },
-    { type: "testing", items: [] },
-    { type: "lock", items: [] },
-    { type: "misc", items: [] },
-  ];
-
-  const groupedLinks = actionOptions.reduce((groups, option) => {
-    // We don't include the delete action in the VM table because it is handled
-    // separately
-    if (appearance === "vmTable" && option.name === NodeActions.DELETE) {
-      return groups;
-    }
-    const count = machines.reduce((sum, machine) => {
-      if (machine.actions.includes(option.name)) {
-        sum += 1;
-      }
-      return sum;
-    }, 0);
+  return actionGroups.reduce<ActionLink[][]>((links, group) => {
+    const groupLinks = group.actions.reduce<ActionLink[]>(
+      (groupLinks, action) => {
+        // We don't include the delete action in the VM table because it is
+        // handled separately
+        if (appearance === "vmTable" && action.name === NodeActions.DELETE) {
+          return groupLinks;
+        }
+        const count = machines.reduce(
+          (sum, machine) =>
+            machine.actions.includes(action.name) ? sum + 1 : sum,
+          0
+        );
+        // Lifecycle actions get displayed regardless of whether the selected
+        // machines can perform them. All other actions are not shown.
```

Comment:
I suspect this will get revisited when we adapt this to work with Devices and Controllers.

Reply:
Yeah I imagine we'll end up creating a `base` version of `TakeActionMenu` that'll just be presentational and work across models